contracts/proxy/contract.go

```diff
@@ -40,10 +73,128 @@ func Update(nefFile, manifest []byte, data any) {
 	runtime.Log("proxy contract updated")
 }
 
+const submitObjectPutMethod = "submitObjectPut"
+
 // Verify checks whether carrier transaction contains either (2/3N + 1) or
 // (N/2 + 1) valid multi-signature of the NeoFS Alphabet.
+// Container contract's `SubmitObjectPut` is an exception, Alphabet signature
+// is not required, contract's `VerifyPlacementSignatures` is called instead.
 func Verify() bool {
-	return common.ContainsAlphabetWitness()
+	if common.ContainsAlphabetWitness() {
+		return true
+	}
+
+	// Container's `SubmitObjectPut` case is only expected then
+
+	script := runtime.GetScriptContainer().Script
+	var i int
+	var sigsLen int
+	var sigs []interop.Signature
+	for i < len(script) {
+		op := script[i]
+		var done bool
+
+		switch {
+		case op == 0x0C: // PUSHDATA1
+			i++
+			if script[i] != interop.SignatureLen {
+				panic("invalid signature length " + std.Itoa(int(script[i]), 10))
+			}
+			i++
+			sigs = append(sigs, script[i:i+interop.SignatureLen])
+			i += interop.SignatureLen
+		case op <= 0x05: // PUSHINT8 to PUSHINT256
+			lenInBytes := int(op + 1)
+			i++
+			sigsLen = convert.ToInteger(script[i : i+lenInBytes])
+			i += lenInBytes
+			done = true
+		case 0x10 < op && op <= 0x20: // PUSH1 to PUSH16
+			sigsLen = int(op - 0x10)
+			i++
+			done = true
+		default:
+			panic("invalid op in signatures " + std.Itoa(int(op), 16))
+		}
+
+		if done {
+			break
+		}
+	}
+	if sigsLen != len(sigs) {
+		panic("number of signatures and pack opcode do not match")
+	}
+	if script[i] != 0xC0 { // PACK
+		panic("signatures array does not start with PACK")
+	}
+	i++
+
+	op := int(script[i])
+	i++
+	var metaLenInBytes int
+	switch op {
+	case 0x0C: // PUSHDATA1
+		metaLenInBytes = 1
+	case 0x0D: // PUSHDATA2
+		metaLenInBytes = 2
+	case 0x0E: // PUSHDATA4
+		metaLenInBytes = 4
+	default:
+		panic("invalid meta push opcode " + std.Itoa(op, 16))
+	}
+	var metaLen int
+	//nolint:intrange
+	for j := 0; j < metaLenInBytes; j++ {
+		metaLen += int(script[i]) << (8 * j)
```

Comment:
and one more non-stylish push: i still think that is the only way to parse byte array length, `convert.ToInteger` works differently, @AnnaShaleva, @roman-khimov, validate, please